paasta_itests/steps/paasta_deployd_steps.py

```diff
@@ -52,16 +57,15 @@ def stop_deployd(context):
 @then('a second deployd does not become leader')
 def start_second_deployd(context):
     context.daemon1 = Popen('paasta-deployd', stderr=PIPE)
-    output = context.daemon1.stderr.readline()
+    output = context.daemon1.stderr.readline().decode('utf-8')
     fd = context.daemon1.stderr
     fl = fcntl.fcntl(fd, fcntl.F_GETFL)
     fcntl.fcntl(fd, fcntl.F_SETFL, fl | os.O_NONBLOCK)
     for i in range(0, 5):
         try:
-            while True:
```

Comment:
Gosh, how did this ever work before? Surely the while loop never exited but I'm pretty sure the tests were passing.